flink-connectors/flink-connector-aws-kinesis-data-streams/src/main/java/org/apache/flink/connector/kinesis/sink/KinesisDataStreamsSinkBuilder.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.kinesis.sink;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.connector.base.sink.AsyncSinkBaseBuilder;
+
+import software.amazon.awssdk.services.kinesis.model.PutRecordsRequestEntry;
+
+import java.util.Properties;
+
+/**
+ * Builder to construct {@link KinesisDataStreamsSink}.
+ *
+ * <p>The following example shows the minimum setup to create a {@link KinesisDataStreamsSink} that
+ * writes String values to a Kinesis Data Streams stream named your_stream_here.
+ *
+ * <pre>{@code
+ * KinesisDataStreamsSink<String> kdsSink =
+ *                 KinesisDataStreamsSink.<String>builder()
+ *                         .setElementConverter(elementConverter)
```

Comment:
I am also wondering whether it is really possible to use any `ElementConverter` or should the sink enforce to always use the `KinesisDataStreamsSinkElementConverter`

Reply:
I think it would be a good idea to let power users create their own `PutRecordRequestEntry` since there are some lesser know and lesser used methods in there like `explicitHashKey` and others that the `KinesisDataStreamsSinkElementConverter` doesn't use.